ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -71,44 +75,105 @@ class MainActivity : ComponentActivity() {
+        //
+        // We also use these scrolling states for various other things (like hiding the time
+        // when the user is scrolling and only showing the vignette when the screen is
+        // scrollable).
+        //
+        // Remember, mobile guidelines specify that if you back navigate out of a screen and then
+        // later navigate into it again, it should be in its initial scroll state (not the last
+        // scroll location it was in before you backed out).
+        val currentBackStackEntry by swipeDismissableNavController.currentBackStackEntryAsState()
+
+        val scrollType =
+            currentBackStackEntry?.arguments?.getSerializable(SCROLL_TYPE_NAV_ARGUMENT)
+                ?: DestinationScrollType.NONE
 
         Scaffold(
-            // Scaffold places time at top of screen to follow Material Design guidelines.
             timeText = {
+                // Scaffold places time at top of screen to follow Material Design guidelines.
+                // (Time is hidden while scrolling.)
+                val activelyScrolling =
+                    when (scrollType) {
+                        DestinationScrollType.SCALING_LAZY_COLUMN_SCROLLING -> {
+                            val viewModel: ScalingLazyListStateViewModel =
+                                viewModel(currentBackStackEntry!!)
+                            viewModel.scrollState.isScrollInProgress
+                        }
+                        DestinationScrollType.COLUMN_SCROLLING -> {
+                            val viewModel: ScrollStateViewModel =
+                                viewModel(currentBackStackEntry!!)
+                            viewModel.scrollState.isScrollInProgress
+                        }
+                        else -> {
+                            false
+                        }
+                    }
+
                 CustomTimeText(
-                    visible = !scalingLazyListState.isScrollInProgress,
+                    visible = !activelyScrolling,
                     showLeadingText = showProceedingTextBeforeTime,
                     leadingText = stringResource(R.string.leading_time_text)
                 )
             },
             vignette = {
-                CustomVignette(
-                    visible = vignetteVisible,
-                    vignettePosition = VignettePosition.TopAndBottom
-                )
+                // Only show vignette for screens with scrollable content.
+                when (scrollType) {
+                    DestinationScrollType.SCALING_LAZY_COLUMN_SCROLLING -> {
+                        CustomVignette(
+                            visible = vignetteVisiblePreference,
+                            vignettePosition = VignettePosition.TopAndBottom
+                        )
+                    }
+                    DestinationScrollType.COLUMN_SCROLLING -> {
+                        CustomVignette(
+                            visible = true,
```

Comment:
Should this be vignetteVisiblePreference?

Reply:
I was initially keeping them separate so it just showed on the list page, but I can see what you mean, i.e., it may be more confusing why they are separated. I combined them to simplify them, thanks!